common/sync.go

```diff
@@ -0,0 +1,5 @@
+package common
+
+import "sync"
+
+var Mu sync.Mutex
```

Comment:
If we are going to have mutexes in common it might be better to have target mutexes - for example in the current situation a mutex for the stream settings. If we reuse the same mutex across the whole app we could be locking resources unrelated to each other. What do you think about having a MutexManager? Something like:
`
type MutexManager struct {
    StreamSettings sync.Mutex
}
`
and initialize and use that

Reply:
Well... the way I implemented the common mutex could really cause us some issues in the future. So, your suggestion is to define the MutexManager (renaming the sync.go to mutex_manager.go) and use its mutexes in the whole application, right?

Reply:
The refactor is available [in this commit](https://github.com/shamanec/GADS/pull/145/commits/8ada1eebe18344f4cf6e0d465c0e8f7b7057257a)

Reply:
Yeah, I guess having them centralized would be better in the cases it makes sense 😅 now there are tons of them here and there